src/mumble_exe/mumble_exe.cpp

```diff
@@ -125,8 +137,151 @@ static const std::wstring GetAbsoluteMumbleAppDllPath(std::wstring suggested_bas
+	// to use a buffer of 7 elements in order to query for the string "false".
+	//
+	// If the value's type is string, we'll end up with "false\0\0", because we
+	// always zero-pad.
+	//
+	// But that's better than ending up in an infinite loop because we failed
+	// to zero-terminate.
+	wchar_t buf[7];
+	memset(&buf, 0, sizeof(buf));
+	DWORD sz = sizeof(buf) - 1*sizeof(wchar_t);
+
+	HKEY key = NULL;
+	bool success = (RegOpenKeyExW(HKEY_CURRENT_USER, L"Software\\Mumble\\Mumble\\shortcuts\\windows\\uiaccess", NULL, KEY_READ, &key) == ERROR_SUCCESS) &&
+	               (RegQueryValueExW(key, L"enable" , NULL, NULL, (LPBYTE)&buf, &sz) == ERROR_SUCCESS);
+	if (success && _wcsicmp(buf, L"false") == 0) {
+		return true;
+	}
+
+	return false;
+}
+
+// SystemSupportsUIAccess returns true if the
+// operating system supports UIAccess.
+static bool SystemSupportsUIAccess() {
+	OSVERSIONINFOEXW ovi;
+	memset(&ovi, 0, sizeof(ovi));
+
+	ovi.dwOSVersionInfoSize = sizeof(ovi);
+	GetVersionEx(reinterpret_cast<OSVERSIONINFOW *>(&ovi));
+
+	// Windows Vista and later support UIAccess.
+	return ovi.dwMajorVersion > 5;
+}
+
+// ProcessHasUIAccess returns true if the current process has UIAccess enabled.
+static bool ProcessHasUIAccess() {
+	HANDLE token = NULL;
+	if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &token)) {
+		return false;
+	}
+
+	DWORD ui_access = 0;
+	DWORD ui_access_size = sizeof(ui_access);
+	if (!GetTokenInformation(token, TokenUIAccess, &ui_access, sizeof(ui_access), &ui_access_size)) {
+		return false;
+	}
+
+	return ui_access == 1;
+}
+
+// RelaunchWithoutUIAccessIfNecessary relaunches the process
+// without UIAccess, if neccs.
+static bool RelaunchWithoutUIAccessIfNecessary() {
+	if (!SystemSupportsUIAccess()) {
+		return true;
+	}
+	if (!ProcessHasUIAccess()) {
+		return true;
+	}
+	if (!UIAccessDisabledViaConfig()) {
+		return true;
+	}
+
+	STARTUPINFO startup_info;
+	memset(&startup_info, 0, sizeof(startup_info));
+
+	GetStartupInfo(&startup_info);
+
+	PROCESS_INFORMATION process_info;
+	memset(&process_info, 0, sizeof(process_info));
+
+	// Store the original value of __COMPAT_LAYER so we can
+	// restore it (which may be equivalent to not setting it all)
+	// once we've been relaunched.
+	wchar_t *compat_layer_env = _wgetenv(L"__COMPAT_LAYER");
+	if (compat_layer_env == NULL) {
+		compat_layer_env = L"mumble_exe_none";
+	}
+	_wputenv_s(L"mumble_exe_prev_compat_layer", compat_layer_env);
+
+	// The only way we have found thus far to ignore the executable's
+	// manifest when calling CreateProcess, is to use the __COMPAT_LAYER
+	// envioronment variable.
+	//
+	// Setting this to RunAsInvoker seemingly ignores the application's
+	// manifest. RunAsInvoker is the default execution level for Windows
+	// programs.
+	_wputenv_s(L"__COMPAT_LAYER", L"RunAsInvoker");
+
+	std::wstring exe_path = GetExecutablePath();
+	if (exe_path.empty()) {
+		return false;
+	}
+
+	if (!CreateProcessW(exe_path.c_str(),
+	              GetCommandLineW(),
```

Comment:
I don't think this is correct in this configuration. ```GetCommandLine``` also included the program call itself in my local tests. Passing it to ```lpApplicationName``` and leaving ```lpCommandLine``` as ```NULL``` might be correct but I'm not sure. It feels quite dangerous to pass it that way given how creative windows can get with that argument...